Core/include/Acts/Utilities/TransformRange.hpp

```diff
@@ -0,0 +1,228 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+namespace Acts {
+
+namespace detail {
+
+template <typename Callable, typename iterator_t, bool force_const>
```

Comment:
One last thing that came to mind: Since the ```TransformRange``` name seems to suggest it is quite generic, why not make ```Callable``` a functor, instead of ducktyping the ```apply``` method and ```value_type```? You could then do:
```
std::invoke_result_t<Callable, internal_value_type>
```
or
```
std::invoke_result_t<Callable, decltype(std::declval<internal_value_type>())>
```

Reply:
And then create `Callable{}` on the fly when I need to invoke it?

Reply:
Or hold it as a member and use ```std::invoke```? Idk, just a suggestion

Reply:
I think I'll keep `apply` for right now.